src/odemis/acq/fastem.py

```diff
@@ -645,13 +645,19 @@ def acquire_roa(self, dataflow):
             # TODO remove the below temporary fix when a proper solution is found
             # The temporary fix has been applied so that the ROAs acquisition can still continue upon failure
             # of correcting the beam shift
-            try:
-                self.correct_beam_shift()  # correct the shift of the beams caused by the parasitic magnetic field.
-            except Exception:
-                logging.exception("Correcting the beam shift failed, check if the image quality is still good.")
-                # In case of failure save the ccd image
-                ccd_image = self._ccd.data.get(asap=False)
-                fastem_util.save_image(self.path, f"{self.field_idx}_after.tiff", ccd_image)
+            # field indices are saved (col, row), run the beamshift correction for every 4th column.
+            # TODO value 4 needs to be updated based on experimental testing
+            if field_idx[0] % 4 == 0 or bs_fail:
+                logging.debug(f"Will run beamshift correction for field index {field_idx}")
```

Comment:
Please spell "beam shift", with or without a space, the same way everywhere (it's easier when searching the logs).